### DISCUSSION TITLE
More discussion on the broken  CUSTOM CODE block

### DISCUSSION THREAD

**Discussion by scottw12** [vanjarosoftware/Vanjaro.Platform#1135] [Q&A] [answered]
So I am still trying to find a way to get certain dynamic things display on a page....

I have a BRAND NEW fresh install with several pages with only a global header and a global footer. 
 
my steps to reproduce error:
1. Create a new BLANK page.
2. Drag a Custom Code Block.
3. Add this code to the block : <p>© Copyright 2017 <script>new Date().getFullYear()>2017&&document.write("- "+new Date().getFullYear());</script> Company Name – All rights reserved.</p>

as soon the page refreshes on publish I get this:
![image](https://user-images.githubusercontent.com/6786299/116458107-d70cd500-a829-11eb-9688-2348d6816544.png)

Live site looks like this after I refresh (which is correct):
![image](https://user-images.githubusercontent.com/6786299/116458288-176c5300-a82a-11eb-829b-88d9e5d3bb11.png)

So  some one tell my why CUSTOM CODE always give me these problems.... Last time I tried an inline Java it destroyed an entire beta site until I figured out how to get into the database itself to remove it. BEWARE using  "Custom Code" in a Global Block... 

I made the page into a template and exported it: 
[test.zip](https://github.com/vanjarosoftware/Vanjaro.Platform/files/6394244/test.zip)

**Comment by meetmandeep** [comment, marked as answer]
> So some one tell my why CUSTOM CODE always give me these problems....

In this particular case, it's the JavaScript itself. 

`document.write("- "+new Date().getFullYear());` is replacing the entire page.

See this: https://stackoverflow.com/questions/19941866/document-write-overwriting-the-document

I've tested the following code in RC 2 Nightly Build in Custom Block and it works great (within global block as well). 

```
© Copyright 2017 - 
<span id="currentYear">2017</span> Company Name – All rights reserved.
<script>
  document.getElementById("currentYear").innerHTML=new Date().getFullYear();
</script> 
```

**Reply by scottw12** [reply]
You have my permission to just bitch slap me LOL...
I am really not that dumb, but boy ohh boy this was plain ole dumb wasn't it.

Anyway no more custom code blocks until RC2 comes out :-)  ...  I
